packages/lib-utils/src/k8s/k8s-utils.ts

```diff
@@ -50,3 +62,189 @@ export const getK8sResourceURL = (
 
   return `/${resourcePath}`;
 };
+
+const requirementToString = (requirement: MatchExpression): string => {
+  if (requirement.operator === 'Equals') {
+    return `${requirement.key}=${requirement.values?.[0]}`;
+  }
+
+  if (requirement.operator === 'NotEquals') {
+    return `${requirement.key}!=${requirement.values?.[0]}`;
+  }
+
+  if (requirement.operator === 'Exists') {
+    return requirement.key;
+  }
+
+  if (requirement.operator === 'DoesNotExist') {
+    return `!${requirement.key}`;
+  }
+
+  if (requirement.operator === 'In') {
+    return `${requirement.key} in (${_.toArray(requirement.values).join(',')})`;
+  }
+
+  if (requirement.operator === 'NotIn') {
+    return `${requirement.key} notin (${_.toArray(requirement.values).join(',')})`;
+  }
+
+  if (requirement.operator === 'GreaterThan') {
+    return `${requirement.key} > ${requirement.values?.[0]}`;
+  }
+
+  if (requirement.operator === 'LessThan') {
+    return `${requirement.key} < ${requirement.values?.[0]}`;
+  }
+
+  return '';
+};
+
+const createEquals = (key: string, value: string): MatchExpression => ({
+  key,
+  operator: 'Equals',
+  values: [value],
+});
+
+const isOldFormat = (selector: Selector | MatchLabels) =>
+  !selector.matchLabels && !selector.matchExpressions;
+
+const toRequirements = (selector: Selector = {}): MatchExpression[] => {
+  const requirements: MatchExpression[] = [];
+  const matchLabels = isOldFormat(selector) ? selector : selector.matchLabels;
+  const { matchExpressions } = selector;
+
+  Object.keys(matchLabels || {})
+    .sort()
+    .forEach((k) => {
+      const value = matchLabels?.[k] as string;
+      requirements.push(createEquals(k, value));
+    });
+
+  matchExpressions?.forEach((me) => {
+    requirements.push(me);
+  });
+
+  return requirements;
+};
+
+export const selectorToString = (selector: Selector): string => {
+  const requirements = toRequirements(selector);
+  return requirements.map(requirementToString).join(',');
+};
+
+export const k8sWatch = (
+  kind: K8sModelCommon,
+  query: {
+    labelSelector?: Selector;
+    resourceVersion?: string;
+    ns?: string;
+    fieldSelector?: string;
+  } = {},
+  wsOptions: {
+    [key: string]: unknown;
+  } = {},
+) => {
+  const queryParams: QueryParams = { watch: 'true' };
+  const opts: {
+    queryParams: QueryParams;
+    ns?: string;
+  } = { queryParams };
+  const wsOptionsUpdated = {
```

Comment:
This probably should be typed.
```suggestion
  const wsOptionsUpdated: WebSocketOptions = {
```